src/http/http_sig.h

```diff
@@ -110,13 +110,13 @@ namespace http
       throw std::logic_error("Unable to sign HTTP request");
     }
 
-    const auto signature = kp->sign(to_sign.value(), MBEDTLS_MD_SHA256);
+    const auto signature = kp->sign(to_sign.value());
 
     auto auth_value = fmt::format(
       "Signature "
       "keyId=\"ignored\",algorithm=\"{}\",headers=\"{}\",signature="
       "\"{}\"",
-      auth::SIGN_ALGORITHM_SHA256,
+      auth::SIGN_ALGORITHM_HS_2019,
```

Comment:
I'm surprised that this is the only change around here - where has the actual implementation of the signing changed? I guess this relies on the passed `kp` using HS2019 rather than SHA256 when we call `kp->sign`? Is there a way to request the signing algorithm from the `kp`, to confirm its one of the algorithms we expect?

Reply:
After discussing, the actual change I was looking for here is the removal of `, MBEDTLS_MD_SHA256` from `kp->sign`. This doesn't mean _no_ hashing, it means the hashing algorithm is _determined from the key_, which is the HS2019 behaviour we're reporting here. I'm happy!